### DISCUSSION TITLE
how can i disable cmp auto bracket function arguments?

### DISCUSSION THREAD

**Discussion by richaardev** [LazyVim/LazyVim#3510] [Q&A]
Hi guys, every time I autocomplete, LazyVim automatically makes the insertion of the function parameters as in the print below, I would like to know if there is the possibility to disable this or disable the cmp autobrackets 

![image](https://github.com/LazyVim/LazyVim/assets/16388439/ddf911e3-da37-42c0-a804-5d89606d862e)

**Comment by folke** [comment]
```lua
  -- Not all LSP servers add brackets when completing a function.
  -- To better deal with this, LazyVim adds a custom option to cmp,
  -- that you can configure. For example:
  --
  -- ```lua
  -- opts = {
  --   auto_brackets = { "python" }
  -- }
  -- 
```

**Reply by richaardev** [reply]
Well this does not work as expected.

I want something like this:
`slog.In|` -> hit enter -> `slog.Info(|)` *but without those function parameter names

https://github.com/LazyVim/LazyVim/assets/16388439/6855a0f3-63e1-42d1-9fab-53950dec2a44

**Reply by folke** [reply]
Oh sorry, this has nothing to do with that setting. That's your LSP providing a snippet.
Check your LSP setttings

**Reply by richaardev** [reply]
> Oh sorry, this has nothing to do with that setting. That's your LSP providing a snippet. Check your LSP setttings

I looked to [gopls lsp settings](https://github.com/LazyVim/LazyVim/blob/59495c18714d8ae4c0b5f7a41dad96e39f9eb189/lua/lazyvim/plugins/extras/lang/go.lua#L28-L62) and I guess there is nothing related that I can change, but I also wanted to change this to others languages because this also occurs with typescript

**Reply by dpetka2001** [reply]
There's a chance this can also come from noice.nvim `lsp.signature` feature. Check the [configuration](https://github.com/folke/noice.nvim/#%EF%B8%8F-configuration) what options it takes to disable it in your personal config and see if that's what you want.

**Reply by folke** [reply]
Right, OP is not talking about auto brackets, but indeed about the function signature.
So yes, disable that in noice

**Reply by richaardev** [reply]
I've tried disable that option and also disable the noice plugin but the function parameters continue to appear inside the brackets with the name and type.

What I expect is when I hit enter to auto complete that option, does not show that `msg string, args ...any`

**Reply by folke** [reply]
probably snippets then what you mean? If this is the case, then I don't think you can disable that
Or inlay hints?
You really should be clear on what you mean...

**Reply by dpetka2001** [reply]
Then it's some setting with the LSP server you use and you should read their docs if it can be disabled or not. For example in `lua-language-server` you can disable it with `settings.Lua.completion.callSnippet = "Disable"`.

**Reply by richaardev** [reply]
> probably snippets then what you mean? If this is the case, then I don't think you can disable that Or inlay hints? You really should be clear on what you mean...

yeah it is snippets, I was able to disable through this option:
`capabilities.textDocument.completion.completionItem.snippetSupport = false`
But I need to put every single language to `auto_brackets` to autocomplete only the brackets.
